File: Inventory/Middlewares/Authorization.js



const jwt = require('jsonwebtoken');

const ensureAuthenticated = (req,res,next)=>{

    const auth = req.headers['inventoryauthorization'];

    if(auth=='null'){


        return next(new Error('Unauthorized'));
        // res.status(403).json({message:'Unauthorized'});
    }

    else{

        try{

            const decoded = jwt.verify(auth,process.env.INVENTORY_SECRET);
    
            req.inventory = decoded;

            console.log(decoded);

            const originalUrl = req.headers['x-original-url'];

            if (originalUrl.includes(decoded.api_url)){

                next();

            }
            else{

                return res.status(403).json({ error: 'Invalid API URL. Access denied.' });

            }
    
    
        }catch(err){
    
  
            next(new Error('TOKEN IS EXPIRED'));
            // return res.status(403).json({
            //     message:'TOKEN IS EXPIRED'
            // })
        }

    }
    


};

module.exports = ensureAuthenticated;

File: Walmart/Middlewares/authorization.py
from flask import request, jsonify
from functools import wraps 
import jwt
import os

def ensure_authenticated(func):
    @wraps(func)
    def wrapper(*args, **kwargs):

        auth = request.headers.get('productsauthorization')
        
        if not auth:
            return jsonify({"error": "Unauthorized"}), 403

        try:

            decoded = jwt.decode(auth, os.getenv("INVENTORY_SECRET"), algorithms=["HS256"])
            request.inventory = decoded
            print(decoded)
            original_url = request.headers.get('x-original-url')

            if decoded.get("api_url") not in original_url:
                return jsonify({"error": "Invalid API URL. Access denied."}), 403
            
        except jwt.ExpiredSignatureError:
            return jsonify({"error": "Token is expired"}), 403
        
        except Exception as e:
            return jsonify({"error": str(e)}), 403

        return func(*args, **kwargs)
    
    return wrapper
